### ISSUE TITLE
Too many and fewer worker as possible at the same time

### ISSUE THREAD

**Issue by utzer** [friendica/friendica#9704]
- [x] I have searched open and closed issues for duplicates

### Bug Description

My worker is not running, I have to restart mariadb, http, php-fpm and the daemon every 15 minutes to have the worker run for a few minutes, but then it stops completely. Daemon is running, but no more worker processes in "ps aux". 

Too many and fewer worker as possible at the same time. I get this at the same time. Is this correct? 

```
2020-12-23T22:10:30Z daemon [NOTICE]: Load: 0.61/30 - processes: 200/1/3299 - jpm: 0/63/14 (0:0, 20:0/90, 30:0/2876, 40:0/388, 50:1/145) - maximum: 15/15 [] - {"file":"Worker.php","line":768,"function":"tooMuchWorkers","uid":"d9143d","process_id":87990}
```

```
2020-12-23T22:10:30Z daemon [INFO]: There are fewer workers as possible, fork a new worker. {"active":1,"queues":15} - {"file":"Worker.php","line":772,"function":"tooMuchWorkers","uid":"d9143d","process_id":87990}
2020-12-23T22:10:35Z daemon [INFO]: Forcing cron worker call. {"pid":87993} - {"file":"daemon.php","line":190,"function":null,"uid":"d9143d","process_id":87990}
```

### Platform Info

Friendica Version: current RC

Friendica Source: git

PHP version: PHP 7.4.13

SQL version: mariadb  Ver 15.1 Distrib 10.5.8-MariaDB

**Comment by utzer** [comment]
It seems Friendica can not check for the minimum free memory anymore, I had that set to 300 MB or so. Today I changed this to 0, so to disable the function, since then the daemon worked fine and cleaned most of the queue and created lots of new tasks and worked though those as well.
However, any idea why this changed? I mean why I had to change the setting to disable the check for free memory in the worker settings in /admin/site?

**Comment by tobiasd** [comment]
> It seems Friendica can not check for the minimum free memory anymore,

Yes I'm seeing this as well. My solution is to disable it.  LOAD is constantly way over the limits.

**Comment by utzer** [comment]
> Yes I'm seeing this as well. My solution is to disable it. LOAD is constantly way over the limits.

sorry if sometimes I am annoying, but in this case again it was working and it just stopped for no apparent reason. So I was a bit annoyed by that. :-( 
Now I am just happy that the problem is solved for now.

**Comment by tobiasd** [comment]
I have a frontend max LOAD setting of 8, for the workers it is set to 6 and I see many times a days lots of PHP processes spawned and a LOAD of 20 and above, sometimes reaching 90 ... on 4 CPUs

**Comment by utzer** [comment]
> I have a frontend max LOAD setting of 8, for the workers it is set to 6 and I see many times a days lots of PHP processes spawned and a LOAD of 20 and above, sometimes reaching 90 ... on 4 CPUs

I never saw that, but I guess my server is just better equipped that yours. Can you tell if this is a fault in the code (nonfunctional code) or if that proc thing is missing?

**Comment by tobiasd** [comment]
No, the logs are looking unsuspicious so far to me and it also seems not to be a problem of the database having to do some swapping or so. It just happens ;-)

**Comment by utzer** [comment]
> No, the logs are looking unsuspicious so far to me 

I meant why the detection is not working. I wonder what is the reason for this. Also for me it seems to work on one server but not the other.

**Comment by annando** [comment]
Do you think that we can close it? You wrote that you had some memory limit problems?

**Comment by tobiasd** [comment]
Well Friendica is basically unusable for me at the moment. So whatever _you_ changed during the RC phase made it devastating on my RasPi. But sure, close the issue and make the release if you are happy with the current state...

**Comment by utzer** [comment]
> Do you think that we can close it? You wrote that you had some memory limit problems?

I think Friendica is not able to detect the remaining memory, for me this is a bug, which should be fixed somehow.

**Comment by annando** [comment]
Can you document what you detected, including the log entries?

**Comment by AlfredSK** [comment]
Looks like it's working...

```
2020-12-31T22:06:05Z worker [WARNING]: Minimal memory reached. {"free":505,"memtotal":64233,"limit":"512"} - {"file":"Process.php","line":182,"function":"isMinMemoryReached","uid":"3b5c53","process_id":2073186}
2020-12-31T22:06:05Z worker [WARNING]: Memory limit reached, quitting. [] - {"file":"Worker.php","line":177,"function":"isReady","uid":"3b5c53","process_id":2073186}
```

But the value of `free` is questionable.

```
free -m
              total        used        free      shared  buff/cache   available
Mem:          64233       31008        1659         127       31565       32425
```

The 'free' memory is actually 1659 M and not 505 M. Also shouldn't we rather look for the 'available' memory instaed of the completely 'free' memory. Just wondering...

Edit:
> The 'free' memory is actually 1659 M and not 505 M. 

Okay. That value changes too quickly for me. I am too slow to get this right. :-) The detected values are actually correct.
